sisl/grid.py

```diff
@@ -195,6 +218,76 @@ def interp(self, shape, method='linear', **kwargs):
 
         return grid
 
+    def smooth(self, method='gaussian', r=None, mode='wrap', **kwargs):
+        '''
+        Make a smoother grid by applying a filter.
+
+        Parameters
+        -----------
+        method: {'gaussian', 'uniform'}, optional
+            the type of filter to apply to smoothen the grid.
+        r: float or array-like of float, optional
+            the radius of the filter in Angstrom for each axis.
+            If the method is 'gaussian', this is the standard deviation!
+            
+            If a single float is provided, then the same distance will be used for all axes.
+            
+            The defaults are 0.6 Ang for a gaussian filter and 1.5 Ang for a uniform filter.
+        mode: {'wrap', 'mirror', 'constant', 'reflect', 'nearest'}, optional
+            determines how to compute the borders of the grid.
+            The default is wrap, which accounts for periodic conditions. 
+            
+            See https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_filter.html#scipy.ndimage.gaussian_filter
+        '''
+        
+        if r is None:
+            r = 0.6 if method == 'gaussian' else 1.5
+        # Normalize the radius input to a list of radius
+        if isinstance(r, (int, float)):
+            r = [r]*3
+        
+        # Calculate the size of the kernel in pixels (in case the
+        # gaussian filter is used, this is the standard deviation)
+        size = (r/np.linalg.norm(self.dcell, axis=1)).astype(int)
```

Comment:
`np.linalg.norm` is ridiculously slow, use `sisl.utils.fnorm` instead.

Also, use `astype(np.int32)` for explicitness. Another idea would be to first round, then astype.
What you have is equivalent to truncation.

Reply:
Is there a reason you hardcoded the axis in `fnorm`?

Reply:
No ;)
Not really. I have just committed to you can choose, default to last axis.

Please rebase ;)